R/src.R

```diff
@@ -0,0 +1,307 @@
+  }
+
+  outpack::outpack_init(path, logging_console = FALSE)
+
+  invisible(path)
+}
+
+
+src_migrate_cfg <- function(cfg) {
+  ## Can't do: fields, changelog, tags, remote, vault
+  ##
+  ## of these :
+  ##
+  ## * vault and remote will get supported later, I think
+  ## * fields and tags are obsolete, tags is unused
+  ## * changelog needs a total overhaul
+
+  ## Note that this does not at all preserve the comments, and we
+  ## should direct the user to do that, but we might be able to do
+  ## something on that by the time we use this for real.
+  ret <- list()
+  if (!is.null(cfg$global_resources)) {
+    ret$global_resources <- cfg$global_resources
+  }
+
+  if (!is.null(cfg$database)) {
+    ret$plugins <- list("orderly3.db" = cfg$database)
+  }
+
+  ret
+}
+
+
+src_migrate_src <- function(name, cfg, strict) {
+  ## TODO: not yet handled - changelog (overhauling this),
+  ## environment, fields, readme, secrets,
+  ## tags - some of these we might just have some general "extra
+  ## metadata" field really, but it would also be good to check in
+  ## various locations which of these are really used in a meaningful
+  ## way.
+
+  ## TODO: some control parameter here to tune 'instance' or not
+  ## through views, data, connection; easier once we have variable
+  ## interpolation in orderly3 configuration, plus a helper in the db
+  ## plugin.
+
+  migrate <- list(
+    src_migrate_description,
+    src_migrate_parameters,
+    src_migrate_global_resources,
+    src_migrate_resources,
+    src_migrate_depends,
+    src_migrate_artefacts,
+    src_migrate_db_views,
+    src_migrate_db_data,
+    src_migrate_db_connection,
+    src_migrate_packages,
+    src_migrate_sources,
+    src_migrate_script)
+
+  dat <- orderly:::orderly_recipe$new(name, cfg, TRUE)
+  code <- if (strict) "orderly3::orderly_strict_mode()" else character(0)
+  for (f in migrate) {
+    code <- add_section(code, f(cfg, dat))
+  }
+  list(code = code, script = dat$script)
+}
+
+
+src_migrate_description <- function(cfg, dat) {
+  if (is.null(dat$displayname) && is.null(dat$description) &&
+      is.null(dat$custom)) {
+    return(NULL)
+  }
+  args <- list(display = dat$displayname,
+               long = dat$description,
+               custom = dat$fields)
+  args <- args[!vapply(args, is.null, TRUE)]
+  args_str <- paste(sprintf("\n  %s = %s", names(args),
+                            vapply(args, deparse1, "", width.cutoff = 500)),
+                    collapse = ",")
+  sprintf("orderly3::orderly_description(%s)", args_str)
+}
+
+
+src_migrate_parameters <- function(cfg, dat) {
+  if (is.null(dat$parameters)) {
+    return(NULL)
+  }
+  ## TODO: there's some inconsistency here with pluralisation in
+  ## orderly3, might need some updating.
+  fmt <- "orderly3::orderly_parameters(%s)"
+  args <- sprintf("%s = %s", names(dat$parameters),
+                  vapply(dat$parameters, deparse, ""))
+  sprintf(fmt, paste(args, collapse = ", "))
```

Comment:
I think this is not formatting correctly, we're getting here for `default-param` where `dat$parameters` are

```Browse[2]> dat$parameters
$nmin
$nmin$default
[1] 0.5


$disease
$disease$default
[1] "HepB"
```

```
orderly3::orderly_parameters(nmin = list(default = 0.5), disease = list(default = \"HepB\"))
```